File: netcore/src/Koralium.Transport/QueryResult.cs
using System;
using System.Collections.Generic;
using System.Collections.Immutable;
using System.Linq;

namespace Koralium.Transport
{
    public class QueryResult
    {
        public IQueryable Result { get; }

        public IImmutableList<Column> Columns { get; }

        public IEnumerable<KeyValuePair<string, string>> Metadata { get; }

        public QueryResult(IQueryable result, IImmutableList<Column> columns, IEnumerable<KeyValuePair<string, string>> metadata)
        {
            Result = result;
            Columns = columns;
            Metadata = metadata;
        }
    }
}

File: typescript-client/legacy/src/queryResult.ts

export class Metadata {
  customMetadata: {};

  constructor(customMetadata: {}) {
    this.customMetadata = customMetadata;
  }
}

export class QueryResult {
  rows: Array<{}>;
  metadata: Metadata;

  constructor(rows: Array<{}>, metadata: Metadata) {
    this.rows = rows;
    this.metadata = metadata;
  }
}